php/docker/templates/nginx-vhost-conf.d/80-index.conf

```diff
@@ -1,10 +1,122 @@
 # vim: set ft=nginx:
 
+{{- $stackBackends := list "redis" "memcached" "custom" }}
+{{- $truthStrings := list "true" "on" "yes" -}}
+{{- $keyPrefix := default "nginx-cache:" .Env.STACK_PAGE_CACHE_KEY_PREFIX }}
+{{- $keyUID := default "https$request_method$host$request_uri" .Env.STACK_PAGE_CACHE_KEY_UID }}
+
 location / {
     try_files $uri $uri/ /index.php$is_args$args;
 }
 
 location ~ \.php$ {
+    {{- if has .Env.STACK_PAGE_CACHE_BACKEND $stackBackends }}
```

Comment:
I recommend extracting this into another dir and using include here. It will be much cleaner and it will be easily configurable by the user.

This config related to page cache should stay in `php/docker/templates/nginx-vhost-conf.d/page-cache.d/10-index.conf`.

And here you will have only an include like:
```
location ~ \.php$ {
   include ./page-cache.d/*.conf;
   fastcgi_pass   $upstream;
   ...
```

And this file will be clean. What do you think?